pulsar-broker/src/main/java/com/yahoo/pulsar/broker/service/persistent/PersistentTopic.java

```diff
@@ -1082,7 +1083,19 @@ public void checkGC(int gcIntervalInSeconds) {
         producers.forEach(Producer::checkPermissions);
         subscriptions.forEach((subName, sub) -> sub.getConsumers().forEach(Consumer::checkPermissions));
         checkMessageExpiry();
-        return checkReplication();
+        CompletableFuture<Void> result = new CompletableFuture<Void>();
+        checkReplication().thenAccept(res -> {
+            log.info("Policies updated successfully {}", data);
+            result.complete(null);
+        }).exceptionally(th -> {
+            log.error("Policies update failed {} {}, scheduled retry in {} seconds", data, th.getMessage(),
+                    POLICY_UPDATE_FAILURE_RETRY_TIME_SECONDS, th);
+            brokerService.executor().schedule(this::checkReplication, POLICY_UPDATE_FAILURE_RETRY_TIME_SECONDS,
```

Comment:
Actually, this will only retry 1 time and then give up. Should we keep retrying instead? There might be some intermittent error (eg: failures to write on ZK) that will get fixed after a while.